### PR TITLE
Add doc for hosted mode

### PR DESCRIPTION
**Pull request by zhujian7**
Signed-off-by: zhujian <jiazhu@redhat.com>

issue reference: https://github.com/open-cluster-management-io/ocm/issues/194

### CODE REVIEW

**Comment by mikeshng** [comment]
CC @qiujian16 @morvencao

File: content/en/scenarios/deploy-components-in-hosted-mode.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+3. Get the EXTERNAL_HUB_KUBECONFIG kubeconfig.
+
+    ```shell
+    kind get kubeconfig --name hub --internal > ./.external-hub-kubeconfig
+    ```
+
+4. Switch to management cluster and deploy hub components.
+
+    ```shell
+    kubectl config use-context {management-context}
```
Comment (morvencao) [inline content/en/scenarios/deploy-components-in-hosted-mode.md:68]: s/{management-context}/kind-management/

File: content/en/scenarios/deploy-components-in-hosted-mode.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    export HUB_KUBECONFIG={HUB_KUBECONFIG_PATH}
+    expoet EXTERNAL_MANAGED_KUBECONFIG={MANAGED_KUBECONFIG_PATH}
+    # if they are kind clusters, please also set:
+    kind get kubeconfig --name {kind-hub-cluster-name} --internal > {HUB_KUBECONFIG_PATH}
+    kind get kubeconfig --name {kind-managed-cluster-name} --internal > {MANAGED_KUBECONFIG_PATH}
+    ```
+
+3. Switch to management context and deploy agent components on management cluster.
+    ```shell
+    kubectl config use-context {management-context}
```
Comment (morvencao) [inline content/en/scenarios/deploy-components-in-hosted-mode.md:162]: s/{management-context}/kind-management/

File: content/zh/scenarios/deploy-components-in-hosted-mode.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    cluster-manager-registration-controller-84854b8795-7k9j7   1/1     Running   0          92s
+    cluster-manager-registration-webhook-7c99fd7d75-7xqp4      2/2     Running   0          92s
+    cluster-manager-work-webhook-d9c49d4cc-gf7c9               2/2     Running   0          92s
+    ```
+
+## Deploy klusterlet in hosted mode
+
+1. Make sure there are 3 clusters, management, hub and managed cluster, and the cluster manager is deployed in Default
+   or Hosted mode successfully.
+2. Set the env variable HUB_KUBECONFIG and EXTERNAL_MANAGED_KUBECONFIG.
```
Comment (morvencao) [inline content/zh/scenarios/deploy-components-in-hosted-mode.md:150]: /s/variable/variables/

File: content/en/scenarios/deploy-components-in-hosted-mode.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+1. Make sure there are 3 clusters, management, hub and managed cluster, and the cluster manager is deployed in Default
+   or Hosted mode successfully.
+2. Set the env variable HUB_KUBECONFIG and EXTERNAL_MANAGED_KUBECONFIG.
+
+    ```shell
+    export HUB_KUBECONFIG={HUB_KUBECONFIG_PATH}
+    expoet EXTERNAL_MANAGED_KUBECONFIG={MANAGED_KUBECONFIG_PATH}
+    # if they are kind clusters, please also set:
+    kind get kubeconfig --name {kind-hub-cluster-name} --internal > {HUB_KUBECONFIG_PATH}
+    kind get kubeconfig --name {kind-managed-cluster-name} --internal > {MANAGED_KUBECONFIG_PATH}
```
Comment (morvencao) [inline content/en/scenarios/deploy-components-in-hosted-mode.md:157]: s/{kind-hub-cluster-name}/kind-hub/
s/{kind-managed-cluster-name}/kind-managed/

File: content/en/scenarios/deploy-components-in-hosted-mode.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+    ```
+
+2. Join the managed cluster in hosted mode
+
+    ```yaml
+    kind get kubeconfig --name managed --internal > ./.external-managed-kubeconfig
+    kubectl config use-context kind-management
+    clusteradm join --hub-token <tokenID.tokenSecret> \
+      --hub-apiserver <hub_apiserver_url> \
+      --cluster-name cluster1 \
```
Comment (ycyaoxdu) [inline content/en/scenarios/deploy-components-in-hosted-mode.md:226]: --cluster-name managed

**Comment by zhujian7** [comment]
> I think we can focus on deploy klusterlet in hosted mode

@qiujian16 PTAL, since there are some duplicates between the previous commit and the installation section, I moved the klusterlet mode description into installation, after that, it looks like:

image

image

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**